message PEncryptedContainer {
  optional int32 version = 1;
  optional string group  = 2;
  optional int32 parts   = 3;
  optional int32 part    = 4;
  optional bytes pubkey  = 5;
  optional bytes sym_key = 6; // encrypted Symmetric Key
  optional bytes sym_iv  = 7; // IV for symm encrpytion
  optional bytes container = 8; //encrypted PSignedContainer
}

message PSignedContainer {
  optional PInnerContainer inner_cont = 1;
  optional bytes signature  = 2;
}

message PInnerContainer {
  optional string type     = 1;
  optional bytes message  = 2;
  optional bytes receiver = 3;
  optional int64 timestamp = 4;
}

message PPeerList {
  repeated bytes ip = 1;
}
/* nested types */
//_send:
message PSendCommand {
  optional string receiver = 1;
  optional PInnerContainer cont_to_send = 2;
}